pkg/models/metrics/metrics.go

```diff
@@ -750,7 +750,39 @@ func MonitorAllMetrics(monitoringRequest *client.MonitoringRequestParams, resour
 					go func(metricName string) {
 						queryType, params := AssembleComponentRequestInfo(monitoringRequest, metricName)
 						metricsStr := client.SendMonitoringRequest(client.SecondaryPrometheusEndpoint, queryType, params)
-						ch <- ReformatJson(metricsStr, metricName, map[string]string{"resource_name": monitoringRequest.ComponentName})
+						formattedJson := ReformatJson(metricsStr, metricName, map[string]string{"resource_name": monitoringRequest.ComponentName})
+
+						if metricName == "etcd_server_list" {
```

Comment:
don't hardcode metricName here. I think it's better to break down `func MonitorAllMetrics` into some smaller functions, now it's too long